accounts/static/accounts/js/profile.js

```diff
@@ -0,0 +1,392 @@
+    const csrfToken = data.csrf_token;
+    return csrfToken;
+  } catch (error) {
+    console.error("Error fetching CSRF token:", error);
+    throw error; // Rethrow the error to propagate it
+  }
+}
+// Goes to movie page when user clicks movie poster
+function goToMoviePage(movieId) {
+  window.location.href = "/movie/${movieId}";
+}
+
+// Overlay element for background blur
+const overlay = document.querySelector(".overlay");
+
+/****************************************************
+ *
+ * Remove friend functionality
+ *
+ ***************************************************/
+
+const remove_friend_modal = document.getElementById("remove_friend_modal");
+const remove_friend_name = document.getElementById("remove_friend_name");
+const remove_friend_button = document.getElementById("remove_confirm_button");
+const remove_reject_button = document.getElementById("remove_reject_button");
+const remove_friend_response = document.getElementById(
+  "remove_friend_response"
+);
+
+/**
+ * This function opens/resets the modal for removing a friend
+ * from the users friends list.
+ * @param {String} username the username of the friend that may be removed
+ */
+function openRemoveFriendModal(username) {
+  // Make accept/reject buttons visible
+  remove_friend_button.style.display = "inline-block";
+  remove_reject_button.style.display = "inline-block";
+
+  // Update text with friend's username
+  remove_friend_name.innerHTML =
+    "Would you like to remove " + username + " from your friends list?";
+
+  // Make modal visible, blur background
+  overlay.classList.remove("hidden");
+  remove_friend_modal.style.display = "block";
+
+  // Add proper functionality to button
+  remove_friend_button.onclick = function () {
+    removeFriend(username);
+  };
+}
+
+/**
+ * This function closes the modal for removing a friend from the users friends list.
+ */
+function closeRemoveFriendModal() {
+  // Make modal invisible
+  overlay.classList.add("hidden");
+  remove_friend_modal.style.display = "none";
+}
+
+/**
+ * This function sends a POST request to remove a friend from the users friends list.
+ * @param {String} friend_username
+ */
+async function removeFriend(friend_username) {
+  // Make a POST request to remove friend from friends list
+  const csrfToken = await getCSRFToken();
+  fetch("/accounts/remove/friend/", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      "X-CSRFToken": csrfToken,
+    },
+    body: JSON.stringify({ username: friend_username }),
+  })
+    .then((response) => {
+      // Update response message styling
+      if (response.status == 200) {
+        remove_friend_button.style.display = "none";
+        remove_reject_button.style.display = "none";
+        remove_friend_response.style.color = "green";
+      } else {
+        remove_friend_response.style.color = "red";
+      }
+
+      return response.text();
+    })
+    .then((data) => {
+      // Display response message
```

Comment:
i have the same problem with this as i did with the error one - it's... it's right there!
also i have a feeling you hate inline comments